src/agent/onefuzz/src/expand.rs

```diff
@@ -110,23 +110,15 @@ impl<'a> Expand<'a> {
             PlaceHolder::InputFileSha256.get_string(),
             ExpandedValue::Mapping(Box::new(Expand::input_file_sha256)),
         );
-        values.insert(
-            PlaceHolder::MachineId.get_string(),
-            ExpandedValue::Mapping(Box::new(Expand::machine_id)),
-        );
 
         Self { values }
     }
 
-    // Note: this spawns it's own tokio runtime as it's not really workable to
-    // refactor the get_machine_id into a blockable version, however... as this
-    // is implemented as a ExpandedValue::Mapping, launching a tokio runtime
-    // occurs when users actively attempt to expand {machine_id}.
-    fn machine_id(&self, _format_str: &str) -> Result<Option<ExpandedValue<'a>>> {
-        let rt = tokio::runtime::Runtime::new()?;
-        let machine_id = rt.block_on(get_machine_id())?;
-        let value = machine_id.to_hyphenated().to_string();
-        Ok(Some(ExpandedValue::Scalar(value)))
+    // Must be manually called to enable the use of async library code.
+    pub async fn machine_id(self) -> Result<Expand<'a>> {
```

Comment:
Perhaps this could be done as part of `Expand::new()`, which would be useful for enabling future environment specific options, rather than requiring the callers to have to call .machine_id().await?

Reply:
That was my initial thought, but I avoided suggesting that because I didn't want to force `new()` to itself be async, which would complicate some non-`async` uses of `Expand`. But revisiting the idea, we could just add an _additional_ ctor that _is_ async, and assumes it is being run in a managed task VM context. We could also just impl `Default` to provide the non-`async` ctor.

Let's address that in a follow-up PR.